internal/data/talks.go

```diff
@@ -0,0 +1,58 @@
+package data
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	"github.com/golang-insiders/site/internal/types"
+)
+
+type TalkService struct {
+	DB *sql.DB
+}
+
+func (tm TalkService) Insert(ctx context.Context, t *types.Talk) error {
+	query := `
+		INSERT INTO talks (twitter_username, title, summary, timezone)
+		VALUES ($1, $2, $3, $4)
+		RETURNING id, created_at`
+
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
```

Comment:
What was the need to `WithTimeout` here? Does it take more than 3 seconds to retrieve the Twitter user and other respective fields?

Reply:
I used 'context.WithTimeout` here just in case the database takes too long to fetch the results